File: backend/src/main/java/com/ameri/objects/Beans/MagazineBeans.java
package com.ameri.objects.Beans;

import com.ameri.objects.classes.user.editor.Comment;
import com.ameri.objects.classes.user.editor.EditorAccount;
import com.ameri.objects.classes.user.editor.Subscription;
import java.util.List;

public class MagazineBeans{

    private int magazineRecord;
    private int magazineLikes;
    private String magazineName;
    private List<Comment> commentList;
    private List<Subscription> subscriptionList;
    private List<EditorAccount> editorAccountList;

    {
        magazineLikes = 0;
    }

    public MagazineBeans(int magazineRecord, String magazineName) {
        this.magazineRecord = magazineRecord;
        this.magazineName = magazineName;
    }

    public List<Comment> getCommentList() {
        return commentList;
    }

    public void setCommentList(List<Comment> commentList) {
        this.commentList = commentList;
    }

    public List<Subscription> getSubscriptionList() {
        return subscriptionList;
    }

    public void setSubscriptionList(List<Subscription> subscriptionList) {
        this.subscriptionList = subscriptionList;
    }

    public int getMagazineRecord() {
        return magazineRecord;
    }

    public void setMagazineRecord(int magazineRecord) {
        this.magazineRecord = magazineRecord;
    }

    public String getMagazineName() {
        return magazineName;
    }

    public void setMagazineName(String magazineName) {
        this.magazineName = magazineName;
    }

    public List<EditorAccount> getEditorAccountList() {
        return editorAccountList;
    }

    public void setEditorAccountList(List<EditorAccount> editorAccountList) {
        this.editorAccountList = editorAccountList;
    }

    public int getMagazineLikes() {
        return magazineLikes;
    }

    public void setMagazineLikes(int magazineLikes) {
        this.magazineLikes = magazineLikes;
    }
}

File: frontend/src/objects/classes/beans/editorBeans/MagazineBeans.ts
import {Comment} from "../../usuario/editor/Comment";
import {Subscription} from "../../usuario/editor/Subscription";
import {EditorAccount} from "../../usuario/editor/EditorAccount";

export class MagazineBeans{

  magazineRecord: number;
  magazineLikes: number;
  totalCount: number;
  magazineName: string;
  commentList: Array<Comment>;
  subscriptionList: Array<Subscription>;
  editorAccountList: Array<EditorAccount>;


  constructor(magazineRecord: number, magazineLikes: number, totalCount: number, magazineName: string, commentList: Array<Comment>, subscriptionList: Array<Subscription>, editorAccountList: Array<EditorAccount>) {
    this.magazineRecord = magazineRecord;
    this.magazineLikes = magazineLikes;
    this.totalCount = totalCount;
    this.magazineName = magazineName;
    this.commentList = commentList;
    this.subscriptionList = subscriptionList;
    this.editorAccountList = editorAccountList;
  }
}
